### PR TITLE
Add some steps to the README's Hacking section

### PR DESCRIPTION
**Pull request by thutterer**
While setting up my development environment I found some missing pieces for the Hacking section in the README:

- how to install vagrant libvirt plugin
- how to initialize git submodules
- how to restart hawk-development webUI in case it did not come up

### CODE REVIEW

File: README.asciidoc [hunk 1]
```diff
@@ -179,8 +179,21 @@ some steps.
   # vagrant plugin install vagrant-bindfs
 --------------------------------------
 
-This is all you need to prepare initally to set up the vagrant environment,
-now you can simply start the virtual machine with +vagrant up+ and start
+* If you plan to use +libvirt+ as provisioner make sure you have the
```
Comment (tboerger) [inline README.asciidoc]: It's not a provisioner, it's a provider.

File: README.asciidoc [hunk 2]
```diff
@@ -179,8 +179,21 @@ some steps.
...
-This is all you need to prepare initally to set up the vagrant environment,
-now you can simply start the virtual machine with +vagrant up+ and start
+* If you plan to use +libvirt+ as provisioner make sure you have the 
+  libvirt-plugin installed:
+
+--------------------------------------
+  # vagrant plugin install vagrant-libvirt
+--------------------------------------
+
+This is all you need to prepare initally to set up the vagrant environment.
```
Comment (tboerger) [inline README.asciidoc]: This sentence is simply wrong as you have added another step below :)

File: README.asciidoc (hunk 1 above)
Comment (thutterer) [inline README.asciidoc]: Oh yeah... seems I've spent to much time with crowbar :)

File: README.asciidoc (hunk 2 above)
Comment (thutterer) [inline README.asciidoc]: I could argue that it's actually kind of correct as the other step is not vagrant related but I'll reword/reorder it.

**Pull request merged**